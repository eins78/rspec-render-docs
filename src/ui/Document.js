import React from 'react'

// steal a stolen version of github's markdown doc style:
const styleUrl = 'https://cdn.rawgit.com/sindresorhus/github-markdown-css/gh-pages/github-markdown.css'
const inlineStyle = '.markdown-body { box-sizing: border-box; min-width: 200px; max-width: 980px; margin: 0 auto; padding: 45px;}'

const Document = ({ pageTitle, children }) => (
  <html>
    <head>
      <meta charSet='utf-8' />
      <title>{pageTitle}</title>
      <link rel='stylesheet' href={styleUrl} />
      <style>
        {inlineStyle}
      </style>
    </head>
    <body className='markdown-body'>{children}</body>
  </html>
)

export default Document
