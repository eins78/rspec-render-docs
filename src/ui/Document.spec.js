/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'

import Document from './Document'

it('renders', () => {
  const component = renderer.create(
    <Document pageTitle='the title'>
      <p>children content</p>
    </Document>
  )
  expect(component.toJSON()).toMatchSnapshot()
})
