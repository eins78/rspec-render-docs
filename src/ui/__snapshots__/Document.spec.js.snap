exports[`test renders 1`] = `
<html>
  <head>
    <meta
      charSet="utf-8" />
    <title>
      the title
    </title>
    <link
      href="https://cdn.rawgit.com/sindresorhus/github-markdown-css/gh-pages/github-markdown.css"
      rel="stylesheet" />
    <style>
      .markdown-body { box-sizing: border-box; min-width: 200px; max-width: 980px; margin: 0 auto; padding: 45px;}
    </style>
  </head>
  <body
    className="markdown-body">
    <p>
      children content
    </p>
  </body>
</html>
`;
